gdal/frmts/fits/fitsdataset.cpp

```diff
@@ -79,20 +108,46 @@ class FITSRasterBand : public GDALPamRasterBand {
 
   friend class  FITSDataset;
 
+  bool               bHaveOffsetScale;
+  double             dfOffset;
+  double             dfScale;
+
+ protected:
+    FITSDataset       *poDS;
```

Comment:
GDALRasterBand has already a `GDALDataset *poDS` declaration. I'm a bit concerned by this shadowing. Generally in other drivers, when you need to FOODataset class instance, you do a static_cast<FOODataset*>(poDS) in the method(s), possibly affected to a local variable, where you need it

Reply:
I must confess ( :) ) that I have copied the analogous line in `geotiff.cpp` ( https://github.com/OSGeo/gdal/blob/ffd558e8881dadfa5b204e15068ff11fc6cf1957/gdal/frmts/gtiff/geotiff.cpp#L1150  ) , as I needed `bNoDataSet` and `dfNoDataValue` defined for the `FITSDataset`.
If there is a better way to do that I will be happy to modify the code... but I need a more extended example for a better understanding ...sorry ... and thanks!

Reply:
Except that geotiff.cpp uses poGDS, not poDS, so there is no shadowing of GDALRasterBand::poDS member.

Reply:
Ok, I've misunderstood the variable names: it's clear now. Thanks.

Reply:
Fixed in https://github.com/OSGeo/gdal/pull/1298/commits/57f87e7cf15b3181d8d6307edeeec7c1446c8402